hooks/open_with_shotgun_launchapp.py

```diff
@@ -43,6 +43,13 @@ def execute(self, published_file, **kwargs):
             context = self.tank.context_from_entity("Task", published_file["task"].get("id"))
         else:
             context = self.tank.context_from_path(path)
+            # In case the path is not relative to the project, or the project has no schema,
+            # try to still get a relevant context from the entity or the project.
+            if not context or not context.project:
+                if published_file.get("entity"):
+                    context = self.tank.context_from_entity_dictionary(published_file["entity"])
+                if not context or not context.project:
```

Comment:
Can this really happen if there is a `published_file["entity"]`?

Reply:
I was not sure, that's why I did that instead of an `else`.